### PR TITLE
feat: expose graphql call with REST API via Sofa

### PR DESCRIPTION
**Pull request by AtelyPham**
Use [Sofa-api](https://www.sofa-api.com/) to generate REST API route through GraphQL Schema.

Read about the schema and API document in `schema.gql` and `swagger.yml`.

Thanks for previewing 🤗❤️.

Screen Shot 2021-09-21 at 21 33 14

### CODE REVIEW

File: backend/src/main.ts [hunk 1]
```diff
@@ -11,5 +14,35 @@ async function bootstrap() {
...
+  const { schema } = app.get(GraphQLSchemaHost);
+  const openApi = OpenAPI({
+    schema,
+    info: {
+      title: "NFT4Charity API",
+      version: "3.0.0",
+    },
+    servers: [
+      {
+        url: "http://localhost:8080/api",
```
Comment (coleslaw) [inline backend/src/main.ts]: change this to `/rest` nhé?

File: backend/src/main.ts [hunk 2]
```diff
@@ -11,5 +14,35 @@ async function bootstrap() {
...
+      },
+    ],
+  });
+
+  app.use(
+    "/api",
```
Comment (coleslaw) [inline backend/src/main.ts]: `/rest`

File: backend/src/users/user.schema.ts [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
 
 @Schema()
+@ObjectType()
 export class User extends Document {
   @Prop({ required: true })
+  @Field()
   name: string;
 
   @Prop({ required: true })
+  @Field()
```
Comment (coleslaw) [inline backend/src/users/user.schema.ts:12]: Sort annotations Alphabetically please.

**Review by coleslaw — CHANGES_REQUESTED** [review]
You forgot to commit `schema.gql` and `swagger.yml` ?

Remove `yarn.lock`

**Comment by AtelyPham** [comment]
`schema.gql` and `swagger.yml` are auto-generated files, so I ignore those files 😂.

File: backend/src/main.ts (hunk 1 above)
Comment (AtelyPham) [reply backend/src/main.ts]: Updated!
Screen Shot 2021-09-21 at 22 27 58

Comment (AtelyPham) [reply backend/src/main.ts]: Without `yarn.lock`, an error occurs when building image.

Screen Shot 2021-09-21 at 22 33 26

**Pull request closed without merging**